synthtool/languages/python.py

```diff
@@ -93,6 +94,50 @@ def _get_sample_readme_metadata(sample_dir: Path) -> dict:
     return sample_metadata
 
 
+def detect_versions(path: str = "./google/cloud") -> List[str]:
```

Comment:
How hard would it be for us to pull this into a common file, if the logic is identical?